core/templates/pages/diagnostic-test-player-page/diagnostic-test-topic-state.model.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Diagnostic test topic state model.
+ */
+
+import { Question } from 'domain/question/QuestionObjectFactory';
+
+export interface SkillIdToQuestions {
+  [skillId: string]: Question[];
+}
+
+export class DiagnosticTestTopicStateData {
+  // The field keeps track of the number of attempted questions on a topic.
+  _numberOfAttemptedQuestion: number;
```

Comment:
Should be plural ("...Questions").

Reply:
Removed, as the engine service will keep the record of the total number of questions.